primitives/src/merkle_tree/field_based_mht/path.rs

```diff
@@ -375,6 +415,104 @@ impl<T: FieldBasedMerkleTreeParameters> PartialEq<FieldBasedMHTPath<T>>
+                        .get_raw_path()
+                        .iter()
+                ) 
+        {
+            // 'nodes' must be a single element vector equal to 'sibling'
+            if nodes.len() != 1 && sibling != &nodes[0] {
+                return false;
+            }
+            
+            // 'position' must be either 0 or 1 and equal (in boolean) to 'direction'
+            if &(*direction as usize) != position {
+                return false;
+            }
+        }
+
+        // Paths are the same
+        true
     }
 }
+
+#[cfg(all(test, feature = "tweedle"))]
+mod test {
+    use std::convert::TryInto;
+
+    use crate::{
+        crh::{TweedleFrBatchPoseidonHash, TweedleFrPoseidonHash},
+        merkle_tree::TWEEDLE_DEE_MHT_POSEIDON_PARAMETERS, FieldBasedMerkleTreeParameters, FieldBasedMerkleTreePrecomputedZeroConstants, BatchFieldBasedMerkleTreeParameters, FieldBasedMHTPath, FieldBasedBinaryMHTPath, FieldBasedMerkleTreePath,
+    };
+    use algebra::fields::tweedle::Fr;
+    use rand::SeedableRng;
+    use rand_xorshift::XorShiftRng;
+
+    #[derive(Clone, Debug)]
+    struct TweedleFrFieldBasedMerkleTreeParams;
+    impl FieldBasedMerkleTreeParameters for TweedleFrFieldBasedMerkleTreeParams {
+        type Data = Fr;
+        type H = TweedleFrPoseidonHash;
+        const MERKLE_ARITY: usize = 2;
+        const ZERO_NODE_CST: Option<
+            FieldBasedMerkleTreePrecomputedZeroConstants<'static, Self::H>,
+        > = Some(TWEEDLE_DEE_MHT_POSEIDON_PARAMETERS);
+    }
+    impl BatchFieldBasedMerkleTreeParameters for TweedleFrFieldBasedMerkleTreeParams {
+        type BH = TweedleFrBatchPoseidonHash;
+    }
+
+    type TestPath = FieldBasedMHTPath<TweedleFrFieldBasedMerkleTreeParams>;
+    type TestBinaryPath = FieldBasedBinaryMHTPath<TweedleFrFieldBasedMerkleTreeParams>;
+
+    #[test]
+    fn test_path_eq_operators_tweedle_fr() {
+
+        let rng= &mut XorShiftRng::seed_from_u64(1231275789u64);
+
+        let height1 = 10;
+        let height2 = 11;
+
+        // Equality sanity checks on FieldBasedMHTPath
+        let path1 = TestPath::random(rng, height1);
+        let path2 = TestPath::random(rng, height1);
+        let path3 = TestPath::random(rng, height2);
+        
+        assert_eq!(path1, path1);
+        assert_ne!(path1, path2);
+        assert_ne!(path1, path3);
+        assert_ne!(path2, path3);
+
+        // Equality sanity checks on FieldBasedBinaryMHTPath
+        let binary_path1: TestBinaryPath = path1.clone().try_into().unwrap();
+        let binary_path2: TestBinaryPath = path2.clone().try_into().unwrap();
+        let binary_path3: TestBinaryPath = path3.clone().try_into().unwrap();
+        
+        assert_eq!(binary_path1, binary_path1);
+        assert_ne!(binary_path1, binary_path2);
+        assert_ne!(binary_path1, binary_path3);
+        assert_ne!(binary_path2, binary_path3);
+
+        // Mixed equality operator sanity checks
+        assert_eq!(binary_path1, path1);
```

Comment:
Just to rule out weird behaviors, I would add also a further check that inequality on equal path works (and similarly equality on different paths) by adding the following assertions:
`assert_eq!(binary_path2 != path2, false);`
`assert_eq!(binary_path1 == path2, false);`

Reply:
Done